### DISCUSSION TITLE
AMP Project

### DISCUSSION THREAD

**Discussion by marceloogeda** [frontendbr/forum#175] [Perguntas e Respostas]
Fala galera, tenho acompanhado o desenvolvimento do [AMP](https://github.com/ampproject) mas ainda não tive oportunidade de utiliza-lo. Alguém conhece ou já utilizou em algum projeto? Poderia compartilhar a experiência?

**Comment by tarciozemel** [comment]
Conhecer eu conheço -- cheguei a escrever [um artigo a respeito](http://desenvolvimentoparaweb.com/miscelanea/accelerated-mobile-pages-amp/) -- e testei numa paginazinha de nada na semana em que ele foi lançado.

Notei que, pelo que dá para se constatar com a experiência do dia a dia, com uma boa configuração de infra e um front-end decente, é possível chegar beeem perto do objetivo o AMP de carregar páginas instantaneamente sem as limitações que ele oferece.

**Comment by vitor-mendrone-deel** [comment]
Li pouco sobre o AMP e não acompanhei muito as notícias. Ele faz basicamente o mesmo que o Facebook Instant Articles, certo? 

Esse pouco me fez entender (talvez erroneamente) que o que o AMP (ou o FIA que citei acima) não fazem nada de novo, ou melhor, não fazem nada que não possamos fazer se reduzirmos em 90% o uso de JS e Imagens nas páginas. 

Acho que isso faz dele uma ótima pedida para blogs e sites de notícias mas o torna péssimo para outros tipos de aplicações (tipo e-commerce, com o qual tenho maior familiaridade)

Se eu estiver errado, me corrijam por favor.

**Comment by kazzkiq** [comment]
@Mendrone Eu penso exatamente como você sobre o AMP. Ele é apenas um guia "pronto" de boas práticas, mas tudo que ele disponibiliza já podia ser alcançado antes, com uma otimização adequada (ok, talvez não seria alcançado 100%, mas daria pra chegar bem próximo dessa otimização). Eu penso que essa foi uma forma de a Google cutucar as grandes empresas que muitas vezes [criticavam](http://www.theverge.com/2015/7/20/9002721/the-mobile-web-sucks) a web pela performance _podre_ na maioria dos sites.

Na minha opinião a Google, com o projeto AMP, meio que deu uma indireta: _"tão vendo aqui? O problema nunca foi a web, mas sim esses seus sites mal feitos. Se a gente acabou de provar que dá fazer sites super-rápidos, de quem será que é a culpa que os seus são uma porcaria?"_

Dessa forma eles mataram 3 coelhos com uma cajadada:
- Demonstraram que o problema nunca foi a web, mas sim a falta de qualidade na construção dos sites mobile;
- Criaram um "padrão" para auxiliar as empresas na criação de sites rápidos de forma simples;
- Agora possuem uma forma mais eficiente de detectar esses sites padronizados, e fica mais fácil dar destaque para eles nas buscas (dando mais acesso as empresas e mais qualidade de conteúdo aos usuários mobile).

**Comment by gss-patricia** [comment]
Estou fazendo uma POC com **React** + **AMP**, o que posso dizer, não está dando "bom". Sou nova com React e AMP, e estou enfrentando um problema, onde o validador AMP reclama que não pode inserir tag **script**, no caso a tag é o bundle.js que o React coloca automaticamente no index.html, bom sendo, assim minha página AMP não é valida.

Penso que qualquer lib que tenha JS não conseguirei utilizar o AMP certinho, como Vue.js, ReactJS, entre outros, pois não é permitido usar JavaScript que não seja do AMP. Por eqto não deu bom, utilizando React. Já para páginas estáticas é excelente neh... rsrsrsrs

**Comment by woliveiras** [comment]
@marceloogeda, implementamos no Viva Decora, da uma olhada nesse artigo: https://vivadecora.github.io/google-amp-como-implementar-paginas-amp.html
